src/raftstore/store/peer.rs

```diff
@@ -187,13 +197,17 @@ pub struct Peer {
     /// delete keys' count since last reset.
     pub delete_keys_hint: u64,
 
-    leader_missing_time: Option<Instant>,
+    pub consistency_state: ConsistencyState,
 
     pub tag: String,
 
     pub last_compacted_idx: u64,
 
+    // if we remove ourself in ChangePeer remove, we should set this flag, then
+    // any following committed logs in same Ready should be applied failed.
+    pending_remove: bool,
     leader_lease_expired_time: Option<Timespec>,
+    leader_missing_time: Option<Instant>,
```

Comment:
related fields should be grouped together, such as `leader_lease_expired_time` and `election_timeout`.